### PR TITLE
Objectio supports mpool

### PR DESCRIPTION
**Pull request by LeftHandCold**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:
Only objectio supports mpool. In addition, tae's blockio will call objectio, which does not support mpool, so a compatibility judgment is made in pr. Next, I will submit a pr that enables blockio to support mpool, and this compatibility judgment will be removed at that time.

### CODE REVIEW

File: pkg/vm/engine/tae/dataio/blockio/reader.go [hunk 1]
```diff
@@ -80,7 +81,7 @@ func (r *Reader) LoadBlkColumns(
 		if err != nil {
 			return bat, err
 		}
-		data, err := col.GetData()
+		data, err := col.GetData(nil)
```
Comment (fengttt) [inline pkg/vm/engine/tae/dataio/blockio/reader.go:84]: nil, will this work at all?

File: pkg/objectio/column.go [hunk 2]
```diff
@@ -183,3 +194,19 @@ func (cb *ColumnBlock) UnMarshalMate(cache *bytes.Buffer) error {
...
 	return err
 }
+
+func (cb *ColumnBlock) freeData(entry fileservice.IOEntry, m *mpool.MPool) {
+	if m != nil {
+		m.Free(entry.Data)
+	}
+}
+
+func (cb *ColumnBlock) allocData(entry fileservice.IOEntry, m *mpool.MPool) (err error) {
```
Comment (fengttt) [inline pkg/objectio/column.go:204]: This code, if m is nil, it will not allocate data?   I don't see how this will work.

if m == nil { panic}?

Comment (LeftHandCold) [reply pkg/objectio/column.go:204]: if m == nil { fileservice will make buffer to fill data}

File: pkg/vm/engine/tae/dataio/blockio/reader.go (hunk 1 above)
Comment (LeftHandCold) [reply pkg/vm/engine/tae/dataio/blockio/reader.go:84]: This pr is to allow objectio to support mpool. As for calling objectio to read and write, it needs to be modified in other pr.

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Review by nnsgmsone — COMMENTED** [review]
These interfaces have nothing to do with cn, right? CN only has interfaces like ReadBlock(blockInfo), GetZonemap(blockInfo), right?

**Review by nnsgmsone — APPROVED** [review]

File: pkg/objectio/column.go (hunk 2 above)
Comment (fengttt) [reply pkg/objectio/column.go:204]: file service will make buffer?   Where is the buffer stored?   How can you tell if you need to free the buffer or not?

**Review by fengttt — APPROVED** [review]

**Comment by LeftHandCold** [comment]
@matrix-meow rerun failed

**Comment by LeftHandCold** [comment]
@matrix-meow rerun failed

**Comment by LeftHandCold** [comment]
@matrix-meow rerun failed

**Comment by LeftHandCold** [comment]
@matrix-meow rerun failed

**Pull request merged**